numpy/core/tests/test_multiarray.py

```diff
@@ -4530,6 +4530,11 @@ def test_mean_values(self):
                 res = _mean(mat, axis=axis) * np.prod(mat.shape)
                 assert_almost_equal(res, tgt)
 
+    def test_mean_float16(self):
+        # This fail if the sum inside mean is done in float16 instead
+        # of float32.
+        assert _mean(np.ones(100000, dtype='float16')) == 1
```

Comment:
Should check that using explicit `dtype` or `out` gives the correct type unless that is checked elsewhere. In fact, it should be checked elsewhere, but may not be.